File: typescript_nestjs/src/core/customer/application/CreateCustomer.ts
import { CustomerMobilePhone } from '../domain/CustomerMobilePhone';
import { CustomerLastName } from '../domain/CustomerLastName';
import { CustomerFirstName } from '../domain/CustomerFirstName';
import { CustomerIdentification } from '../domain/CustomerIdentification';
import { CustomerId } from '../domain/CustomerId';
import { CustomerRepository } from '../domain/CustomerRepository';
import { Customer } from '../domain/Customer';
import { EventBus } from '../../shared/bus/domain/EventBus';


export class CreateCustomer {
    constructor(private customerRepository: CustomerRepository, private eventBus: EventBus) {
    }

    async execute(customerId: CustomerId, identification: CustomerIdentification, firstName: CustomerFirstName, lastName: CustomerLastName, mobilePhone: CustomerMobilePhone): Promise<void> {
        const customer = Customer.create(
            customerId,
            identification,
            firstName,
            lastName,
            mobilePhone,
        );

        await this.customerRepository.store(customer);
        await this.eventBus.publish(customer.pullDomainEvents());
    }
}

File: python_django/core/customer/application/create_customer.py
from core.customer.domain.customer import Customer
from core.customer.domain.customer_id import CustomerId
from core.customer.domain.customer_first_name import CustomerFirstName
from core.customer.domain.customer_repository import CustomerRepository
from core.shared.bus.domain.event_bus import EventBus


class CreateCustomer(object):
    _customer_repository: CustomerRepository
    _event_bus: EventBus

    def __init__(self, customer_repository: CustomerRepository, event_bus: EventBus):
        self._customer_repository = customer_repository
        self._event_bus = event_bus

    def __call__(self, customer_id: CustomerId, customer_name: CustomerFirstName):
        customer = Customer.create(customer_id, customer_name)

        self._customer_repository.store(customer)
        self._event_bus.publish(customer.pull_domain_events())